app/controllers/ops_controller/diagnostics.rb

```diff
@@ -267,15 +267,26 @@ def cu_repair
   end
 
   def db_backup_form_field_changed
+    require 'uri'
     schedule     = MiqSchedule.find_by_id(params[:id])
     depot        = schedule.file_depot
-    uri_settings = depot.try(:[], :uri).to_s.split("://")
+    full_uri, _query = depot.try(:uri)&.split('?')
+    uri_prefix, uri  = full_uri.to_s.split('://')
+    # uri_settings = depot.try(:[], :uri).to_s.split("://")
+    port         = URI(depot.try(:uri)).port
     render :json => {
-      :depot_name     => depot.try(:name),
-      :uri            => uri_settings[1],
-      :uri_prefix     => uri_settings[0],
-      :log_userid     => depot.try(:authentication_userid),
-      :log_aws_region => depot.try(:aws_region)
+      :depot_name           => depot.try(:name),
+      # :uri                  => uri_settings[1],
+      # :uri_prefix           => uri_settings[0],
+      :uri                  => uri,
+      :uri_prefix           => uri_prefix,
+      :log_userid           => depot.try(:authentication_userid),
+      :log_aws_region       => depot.try(:aws_region),
+      :log_openstack_region => depot.try(:openstack_region),
```

Comment:
This should be `:openstack_region` as in js it looks for `data.openstack_region`.